app/models/book.rb

```diff
@@ -1,3 +1,27 @@
 class Book < ApplicationRecord
-  
+
+  has_many :reviews
+  has_many :users, through: :reviews
+
+  def average
+    reviews.average(:rating)
+  end
+
+  def highest_rating
+    reviews.maximum(:rating)
+  end
+
+  def highest_rating_review
+    reviews.find_by(:rating=>4)
+  end
```

Comment:
Here you have hard coded 4 as the highest rating. How could you have built this to be a bit more dynamic?

Reply:
This may have also been creating your Nil class errors. In that test you don't create a review with a rating of 4 so this method likely returned nil. Then when you tried to call user_id off of it, it told you can't call user_id on Nil.